### PR TITLE
Add CONTRIBUTING

### PR DESCRIPTION
**Pull request by codetheweb**
Hey @kueblc, let me know what you think.  I think I covered most of it, but I'm sure there's something I'm missing.

This would get merged into master.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Introduction
+
+Thank you for considering contributing to TuyAPI! We welcome all kinds of contributions, whether it's a small README fix or a new feature.
+
+Please take a movement to read through this short guide so both the core maintainers' time and your time can be better utilized.
```
Comment (kueblc) [inline CONTRIBUTING.md]:
```suggestion
Please take a moment to read through this short guide so both the core maintainers' time and your time can be better utilized.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Before you open a PR, make sure all tests pass (see next section).
+
+## Getting started
+
+You're ready to start programming?  Great, here's what you should do:
+
+1. Fork this repository to your account (click the **Fork** button at the top right).
+2. Clone your fork to your computer.
+3. Run `npm i` to install dependencies.
```
Comment (kueblc) [inline CONTRIBUTING.md]: Create a branch (specify naming scheme)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+4. Make your changes. (Tip: you can use a file named `dev.js` as a local playground to test your new code in, since the file is ignored by Git.)
+5. Ensure test pass with `npm test`. If you get a lot of style-related errors, consider running `npx xo --fix` to automatically fix most of them.
+6. Commit, push, and open a PR. Expect to hear back within a week or so (usually sooner).
```
Comment (kueblc) [inline CONTRIBUTING.md]: Make the PR to `development` not `master`

**Comment by kueblc** [comment]
Looks good!

**Comment by kueblc** [comment]
Maybe now is an appropriate time to add myself to contributors?

**Comment by codetheweb** [comment]
I knew I was forgetting something. 😛

Merging.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Getting started
+
+You're ready to start programming?  Great, here's what you should do:
+
+1. Fork this repository to your account (click the **Fork** button at the top right).
+2. Clone your fork to your computer.
+3. Run `npm i` to install dependencies.
+4. Make a new branch with `git branch`. If you're adding a new feature, use a `feature-*` prefix (ex. `feature-contributing`). If you're fixing a bug, use a `bugfix-*` prefix. Use your best judgement on anything else.
+4. Make your changes. (Tip: you can use a file named `dev.js` as a local playground to test your new code in, since the file is ignored by Git.)
```
Comment (kueblc) [inline CONTRIBUTING.md]: We've got two fours :stuck_out_tongue: but luckily markdown has our back

**Comment by codetheweb** [comment]
Whoops.  Fixed and committed to master.

I need to get some kind of Markdown plugin for Atom that automatically fixes stuff like that.

**Pull request merged**